src/Text.Analyzers/UnitTests/IdentifiersShouldBeSpelledCorrectlyTests.cs

```diff
@@ -1,8 +1,473 @@
+                new object[] { CreateTypeWithEvent("Program", "<?>D"), "D" },
+                new object[] { CreateTypeWithProperty("Program", "<?>E"), "E" },
+                new object[] { CreateTypeWithMethod("Program", "<?>F"), "F" },
+            };
+
+        public static IEnumerable<object[]> MisspelledMemberParameters
+            => new[]
+            {
+                new object[] { CreateTypeWithConstructor("Program", false, "int <?>yourNaem"), "Naem", "yourNaem", "Program.Program(int)" },
+                new object[] { CreateTypeWithMethod("Program", "Method", "int <?>yourNaem"), "Naem", "yourNaem", "Program.Method(int)" },
+                new object[] { CreateTypeWithIndexer("Program", "int <?>yourNaem"), "Naem", "yourNaem", "Program.this[int]" },
+            };
+
+        [Theory]
+        [InlineData("namespace Bar { }")]
+        [InlineData("class Program { }")]
+        [InlineData("class Program { void Member() { } }")]
+        [InlineData("class Program { int _variable = 1; }")]
+        [InlineData("class Program { void Member(string name) { } }")]
+        [InlineData("class Program { delegate int GetNumber(string name); }")]
+        [InlineData("class Program<TResource> { }")]
+        public async Task NoMisspellings_Verify_NoDiagnostics(string source)
+        {
+            await VerifyCSharpAsync(source);
+        }
+
+        [Fact]
+        public async Task MisspellingAllowedByGlobalXmlDictionary_Verify_NoDiagnostics()
+        {
+            var source = "class Clazz { }";
+            var dictionary = CreateXmlDictionary(new[] { "clazz" });
+
+            await VerifyCSharpAsync(source, dictionary);
+        }
+
+        [Fact]
+        public async Task MisspellingAllowedByGlobalDicDictionary_Verify_NoDiagnostics()
+        {
+            var source = "class Clazz { }";
+            var dictionary = CreateDicDictionary(new[] { "clazz" });
+
+            await VerifyCSharpAsync(source, dictionary);
+        }
+
+        [Fact]
+        public async Task MisspellingsAllowedByMultipleGlobalDictionaries_Verify_NoDiagnostics()
+        {
+            var source = @"class Clazz { const string Naem = ""foo""; }";
+            var xmlDictionary = CreateXmlDictionary(new[] { "clazz" });
+            var dicDictionary = CreateDicDictionary(new[] { "naem" });
+
+            await VerifyCSharpAsync(source, new[] { xmlDictionary, dicDictionary });
+        }
+
+        [Fact]
+        public async Task CorrectWordDisallowedByGlobalXmlDictionary_Verify_EmitsDiagnostic()
+        {
+            var source = "class Program { }";
+            var dictionary = CreateXmlDictionary(null, new[] { "program" });
+
+            await VerifyCSharpAsync(
+                source,
+                dictionary,
+                VerifyCS.Diagnostic(IdentifiersShouldBeSpelledCorrectlyAnalyzer.TypeRule)
+                    .WithLocation(1, 7)
+                    .WithArguments("Program", "Program"));
+        }
+
+        // [Fact]
```

Comment:
Tests need to be uncommented or removed.

Reply:
Seems like all diagnostic verification tests are commented out - this will certainly block this PR.

Reply:
At least one of my tests needs to modify the assembly name so it's misspelled/unmeaningful, but it doesn't appear this is possible in [AnalyzerTest`1.cs](https://github.com/dotnet/roslyn-sdk/blob/1f045832e3db8653f9c3b4baecb2d6d133e92096/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest%601.cs#L1100). Is that correct?

I also need `AdditionalFiles` to be associated with specific projects, but they're just listed in [SolutionState.cs](https://github.com/dotnet/roslyn-sdk/blob/master/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/SolutionState.cs) vs. ProjectState.cs.

Reply:
Still working on converting the tests...

Reply:
Tests are "done" (see skipped)